pkg/cnab/config-adapter/adapter_test.go

```diff
@@ -675,6 +675,39 @@ func TestManifestConverter_generateCustomMetadata(t *testing.T) {
 	require.NoError(t, err, "ToBundle failed")
 	assert.Len(t, bun.Custom, 2)
 
-	fooCustomData := bun.Custom["foo"]
-	assert.Equal(t, "bar", fooCustomData)
+	val, ok := bun.Custom["foo"].(map[string]interface{})
```

Comment:
This test should use the original code I pointed you to for Marshaling a bundle to json, not repeat the tests from the manifest test suite. What we need to ensure is that after your change that we have a regression test that checks that we can create bundle.json files that contain nested custom metadata structures.

So let's write the json, compare it to a known string in a testdata file and this should be all set! 👍